src/services/refactors/convertToEs6Module.ts

```diff
@@ -0,0 +1,567 @@
+        switch (name.kind) {
+            case SyntaxKind.ObjectBindingPattern:
+            case SyntaxKind.ArrayBindingPattern: {
+                // `const [a, b] = require("c").d` --> `import { d } from "c"; const [a, b] = d;`
+                const tmp  = makeUniqueName(propertyName, identifiers);
+                return [
+                    makeSingleImport(tmp, propertyName, moduleSpecifier),
+                    makeConst(/*modifiers*/ undefined, name, createIdentifier(tmp)),
+                ];
+            }
+            case SyntaxKind.Identifier:
+                // `const a = require("b").c` --> `import { c as a } from "./b";
+                return [makeSingleImport(name.text, propertyName, moduleSpecifier)];
+            default:
+                Debug.assertNever(name);
+        }
+    }
+
+    function convertAssignment(
+        sourceFile: SourceFile,
+        statement: ExpressionStatement,
+        left: Expression,
+        right: Expression,
+        changes: textChanges.ChangeTracker,
+        exports: ExportRenames,
+    ): ModuleExportsChanged {
+        if (!isPropertyAccessExpression(left)) {
+            return false;
+        }
+
+        if (isExportsOrModuleExportsOrAlias(sourceFile, left)) {
+            if (isExportsOrModuleExportsOrAlias(sourceFile, right)) {
+                // `const alias = module.exports;` or `module.exports = alias;` can be removed.
+                changes.deleteNode(sourceFile, statement);
+            }
+            else {
+                let newNodes = isObjectLiteralExpression(right) ? tryChangeModuleExportsObject(right) : undefined;
+                let changedToDefaultExport = false;
+                if (!newNodes) {
+                    ([newNodes, changedToDefaultExport] = convertModuleExportsToExportDefault(right));
+                }
+                changes.replaceNodeWithNodes(sourceFile, statement, newNodes, { nodeSeparator: "\n", useNonAdjustedEndPosition: true });
+                return changedToDefaultExport;
+            }
+        }
+        else if (isExportsOrModuleExportsOrAlias(sourceFile, left.expression)) {
+            convertNamedExport(sourceFile, statement, left.name, right, changes, exports);
+        }
+
+        return false;
+    }
+
+    /**
+     * Convert `module.exports = { ... }` to individual exports..
+     * We can't always do this if the module has interesting members -- then it will be a default export instead.
+     */
+    function tryChangeModuleExportsObject(object: ObjectLiteralExpression): ReadonlyArray<Statement> | undefined {
+        return mapAllOrFail(object.properties, prop => {
+            switch (prop.kind) {
+                case SyntaxKind.GetAccessor:
+                case SyntaxKind.SetAccessor:
+                // TODO: Maybe we should handle this? See fourslash test `refactorConvertToEs6Module_export_object_shorthand.ts`.

```

Comment:
We could transpile that test to just `export function f() {}` instead of `function f() {} export default { f };`. But it would be work and I don't know how common it is to write a module like that.